### PR TITLE
klll tcomb

### PR DESCRIPTION
**Pull request by jasonLaster**
:gun:  This finally does the messy work of killing tcomb, which was just being used to type chrome.

* There are some places where I skimped on types
* There is some refactoring, which i'll test
* We had basically duped a lot of files for months. yikes

### CODE REVIEW

File: packages/devtools-client-adapters/src/commands.js [hunk 1]
```diff
@@ -1,185 +0,0 @@
-const { BreakpointResult, Location } = require("../tcomb-types");
```
Comment (jasonLaster) [inline packages/devtools-client-adapters/src/commands.js]: 😢

File: packages/devtools-client-adapters/src/chrome.js [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-const { Tab } = require("./tcomb-types");
-
-// TODO: figure out a way to avoid patching native prototypes.
-// Unfortunately the Chrome client requires it to work.
 
-Array.prototype.peekLast = function() { // eslint-disable-line
-  return this[this.length - 1];
-};
+// similar to Tab, but has webSocketDebuggerUrl
+type ChromeTab = {
```
Comment (jasonLaster) [inline packages/devtools-client-adapters/src/chrome.js]: I'd like to extend `Tab` with `webSocketDebuggerUrl`

but i don't think that will work

```
type ChromeTab = {...Tab, webSocketDebuggerUrl: string }
```

**Comment by jasonLaster** [comment]
@clarkbw mind taking a look.

File: packages/devtools-client-adapters/src/chrome.js (hunk 2 above)
Comment (clarkbw) [inline packages/devtools-client-adapters/src/chrome.js]: I think you could do an [intersection type](https://flowtype.org/docs/union-intersection-types.html#_) here. 

Like this?

```js
type ChromeTab = Tab & {webSocketDebuggerUrl: string};
```

File: packages/devtools-client-adapters/src/chrome/commands.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+  breakpointId: string,
+  serverLocation: ServerLocation
+}
 
 let debuggerAgent;
 let runtimeAgent;
 let pageAgent;
 
-function setupCommands({ Debugger, Runtime, Page }) {
+function setupCommands({ Debugger, Runtime, Page }: any) {
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/chrome/commands.js]: Might be good to put this in the local types as well

File: packages/devtools-client-adapters/src/chrome/commands.js [hunk 4]
```diff
@@ -35,49 +45,46 @@ function breakOnNext() {
...
-    location: {
-      scriptId: location.sourceId,
-      lineNumber: location.line - 1
-    },
+async function setBreakpoint(location: Location, condition: string) {
+  let {
+    breakpointId,
+    serverLocation
+  }: setBreakpointResponseType = await debuggerAgent.setBreakpoint({
+    location: toServerLocation(location),
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/chrome/commands.js]: 🎉 so much better!

File: packages/devtools-client-adapters/src/chrome/commands.js [hunk 5]
```diff
@@ -35,49 +45,46 @@ function breakOnNext() {
...
 }
 
-function setBreakpoint(location, condition) {
-  return debuggerAgent.setBreakpoint({
-    location: {
-      scriptId: location.sourceId,
-      lineNumber: location.line - 1
-    },
+async function setBreakpoint(location: Location, condition: string) {
+  let {
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/chrome/commands.js]: `const`?

File: packages/devtools-client-adapters/src/chrome/create.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+// @flow
+
+import type { Location } from "../types";
+import type { ServerLocation } from "./types";
+
+function fromServerLocation(serverLocation: ServerLocation): ?Location {
+  if (serverLocation) {
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/chrome/create.js]: I'm guessing you needed the guard here but it feels like the `serverLocation` isn't optional given how the params are done.

File: packages/devtools-client-adapters/src/defer.js [hunk 7]
```diff
@@ -1,12 +0,0 @@
-module.exports = function defer() {
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/defer.js]: :shipit:

File: packages/devtools-client-adapters/src/firefox/create.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ import type {
   PausedPacket,
   FramesResponse,
   FramePacket,
-  SourcePayload
+  SourcePayload,
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/firefox/create.js]: 😕

File: packages/devtools-client-adapters/src/task.js [hunk 9]
```diff
@@ -1,48 +0,0 @@
-/* -*- indent-tabs-mode: nil; js-indent-level: 2 -*- */
-/* vim: set ts=2 et sw=2 tw=80 filetype=javascript: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-/**
- * This object provides the public module functions.
- */
-const Task = {
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/task.js]: 😍

File: packages/devtools-client-adapters/src/tcomb-types.js [hunk 10]
```diff
@@ -1,59 +0,0 @@
-const t = require("tcomb");
```
Comment (clarkbw) [inline packages/devtools-client-adapters/src/tcomb-types.js]: 🙌

File: packages/devtools-client-adapters/src/chrome/create.js (hunk 6 above)
Comment (jasonLaster) [inline packages/devtools-client-adapters/src/chrome/create.js]: good catch

**Pull request merged**